server/src/main/java/com/exacaster/lighter/backend/yarn/YarnBackend.java

```diff
@@ -100,11 +103,18 @@ public void kill(Application application) {
     }
 
     @Override
-    public Map<String, String> getSubmitConfiguration(Application application,
-            Map<String, String> current) {
+    public SparkApp prepareSparkApplication(Application application, Map<String, String> configDefaults,
```

Comment:
Maybe this could be extracted into CommonUtils? It's the same with KubernetesBackend.